src/main/java/com/redhat/red/build/koji/model/xmlrpc/KojiBuildTypeInfo.java

```diff
@@ -98,90 +94,114 @@ public KojiImageBuildInfo getImage()
     {
         return name;
     }
 
-    public void setName( String name ) {
-        this.name = name;
-    }
-
-    public KojiBuildInfo addBuildTypeInfo( KojiBuildInfo buildInfo )
+    /**
+     * Util method. Add buildTypeInfo to buildInfo.
+     *
+     * @param buildTypeInfo source
+     * @param buildInfo target
+     * @return buildInfo
+     */
+    public static KojiBuildInfo addBuildTypeInfo( KojiBuildTypeInfo buildTypeInfo, KojiBuildInfo buildInfo )
     {
         if ( buildInfo == null )
         {
             return null;
         }
 
-        Object buildTypeInfo = getBuildInfo();
+        List<Object> inf = buildTypeInfo.getBuildInfo();
 
-        if ( buildTypeInfo == null )
+        if ( inf == null || inf.isEmpty() )
         {
             return buildInfo;
         }
 
-        if ( buildTypeInfo instanceof KojiMavenBuildInfo )
+        for ( Object obj : inf )
         {
-            KojiMavenBuildInfo maven = (KojiMavenBuildInfo) buildTypeInfo;
-            buildInfo.setId( maven.getBuildId() );
-            buildInfo.setMavenArtifactId( maven.getArtifactId() );
-            buildInfo.setMavenGroupId( maven.getGroupId() );
-            buildInfo.setMavenVersion( maven.getVersion() );
-        }
-        else if ( buildTypeInfo instanceof KojiWinBuildInfo )
-        {
-            KojiWinBuildInfo win = (KojiWinBuildInfo) buildTypeInfo;
-            buildInfo.setId( win.getBuildId() );
-            buildInfo.setPlatform( win.getPlatform() );
-        }
-        else if ( buildTypeInfo instanceof KojiImageBuildInfo )
-        {
-            KojiImageBuildInfo image = (KojiImageBuildInfo) buildTypeInfo;
-            buildInfo.setId( image.getBuildId() );
+            if ( obj instanceof KojiMavenBuildInfo )
+            {
+                KojiMavenBuildInfo maven = (KojiMavenBuildInfo) obj;
+                buildInfo.setId( maven.getBuildId() );
+                buildInfo.setMavenArtifactId( maven.getArtifactId() );
+                buildInfo.setMavenGroupId( maven.getGroupId() );
+                buildInfo.setMavenVersion( maven.getVersion() );
+            }
+            else if ( obj instanceof KojiWinBuildInfo )
+            {
+                KojiWinBuildInfo win = (KojiWinBuildInfo) obj;
+                buildInfo.setId( win.getBuildId() );
+                buildInfo.setPlatform( win.getPlatform() );
+            }
+            else if ( obj instanceof KojiImageBuildInfo )
+            {
+                KojiImageBuildInfo image = (KojiImageBuildInfo) obj;
+                buildInfo.setId( image.getBuildId() );
+            }
+            else if ( obj instanceof KojiRpmBuildInfo )
+            {
+                KojiRpmBuildInfo rpm = (KojiRpmBuildInfo) obj; // nothing to set
+            }
         }
-
-        buildInfo.setTypeName(name);
+        buildInfo.setTypeName( buildTypeInfo.getName() );
 
         return buildInfo;
     }
 
-    public void setBuildInfo(Object buildInfo)
+    @Deprecated
```

Comment:
This method was never in a release (I didn't see it in 1.9), so it seems we can just remove it from 2.0 if there is no need to keep both.

Reply:
Removed.